code/game/area/Space_Station_13_areas.dm

```diff
@@ -618,10 +618,13 @@ NOTE: there are two lists of areas in the end of this file: centcom and station
 	lighting_colour_bulb = "#ffebc1"
 	sound_environment = SOUND_AREA_WOODFLOOR
 
-/area/crew_quarters/bar/mood_check(mob/living/carbon/human/subject)
-		. = ..()
-		if (HAS_TRAIT(subject, TRAIT_LIGHT_DRINKER))
-				. = FALSE
+/area/crew_quarters/bar/mood_check(mob/living/carbon/subject)
```

Comment:
Why is this being called with null as a parameter in the first place? there is an issue somewhere else

Reply:
well id guess it would be because the proc argument casted it to mob/living/carbon/human? it wouldnt make sense to check for traits on non humans anyway

Reply:
Byond doesn't actually do casting like this, if you pass in a completely different type, if you send in the wrong type as a parameter, the wrong type will get forced into that var and it will only crash when you try to access a variable that exists on the type specified in the parameter but doesn't exist on the type you actually passed in.
The only way for this to be null is if it was passed in as null in the first place.

Reply:
ah found the issue